Compiler/Util/BaseAvlSet.mo

```diff
@@ -134,6 +156,7 @@ algorithm
   key := match inTree
     case EMPTY() algorithm return; then fail();
     case NODE() then inTree.key;
+    case LEAF() then inTree.key;
```

Comment:
Then in (new) line 167 add `else false;` to account for the new record NODE(), otherwise the match fails.